discvrcore/src/org/labkey/discvrcore/AuditSummaryUserSchema.java

```diff
@@ -201,10 +201,10 @@ private Dataset createDataset(String name, boolean isDemographics) throws Except
             Dataset d1 = StudyService.get().createDataset(getContainer(true), TestContext.get().getUser(), name, null, isDemographics);
             d1.setKeyManagementType(Dataset.KeyManagementType.GUID);
             d1.setKeyPropertyName("objectId");
-            DomainProperty objectId1 = d1.getDomain().addProperty();
+            DomainProperty objectId1 = d1.getDomain(true).addProperty();
             objectId1.setName("objectId");
             objectId1.setPropertyURI(AuditSummaryUserSchema.class.getName() + ":ObjectId");
-            d1.getDomain().save(TestContext.get().getUser());
+            d1.getDomain(true).save(TestContext.get().getUser());
```

Comment:
Actually, this looks incorrect. Won't the two calls to `getDomain(true)` return two different objects now? This should hold onto the `Domain` as a local variable and reuse it.